### PR TITLE
Optim-wip: Move inception test helpers to separate file

### PR DESCRIPTION
**Pull request by ProGamerGov**
This change will make it easier to use these helpers for testing other models.

### CODE REVIEW

File: tests/optim/helpers/models.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import torch
+
+
+def _check_layer_in_model(
+    self,
+    model: torch.nn.Module,
+    layer: Type[torch.nn.Module],
+) -> None:
+    def check_for_layer_in_model(model, layer) -> bool:
```
Comment (aobo-y) [inline tests/optim/helpers/models.py:11]: If you agree with the above suggestion, we don't need another nested inner function here.

```py
def check_layer_in_model(model, layer):
  for _, child in model._modules.items():
    if child is None:
      continue
    if isinstance(child, layer) or check_layer_in_model(child, layer):
      return True
  return False
```

File: tests/optim/helpers/models.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                if isinstance(child, layer):
+                    return True
+                if check_for_layer_in_model(child, layer):
+                    return True
+        return False
+
+    self.assertTrue(check_for_layer_in_model(model, layer))
+
+
+def _check_layer_not_in_model(
```
Comment (aobo-y) [inline tests/optim/helpers/models.py:23]: we may not even need this `layer_not_in_model` function, just `not check_layer_in_model(model, layer)`

Comment (ProGamerGov) [reply tests/optim/helpers/models.py:23]: @aobo-y 

In addition to using `not`, we can also use `self.assertFalse`

```
self.assertFalse(check_layer_in_model(model, layer))
self.assertTrue(check_layer_in_model(model, layer))
```

File: tests/optim/helpers/models.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from typing import Type
+
+import torch
+
+
+def _check_layer_in_model(
+    self,
```
[the remark this answers is not in the document]
Comment (ProGamerGov) [reply tests/optim/helpers/models.py:7]: The original idea was to follow after `assertTensorAlmostEqual`'s input variable setup, but it does appear simpler to return bool outputs instead.

File: tests/optim/helpers/models.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+from typing import Type
+
+import torch
+
+
+def _check_layer_in_model(
```
[the remark this answers is not in the document]
Comment (ProGamerGov) [reply tests/optim/helpers/models.py]: Oops, yeah I'm not sure how I missed removing the underscore!

File: tests/optim/helpers/models.py (hunk 1 above)
Comment (ProGamerGov) [reply tests/optim/helpers/models.py]: Yeah, we can just use the one function with bool outputs to test for the presence and absence of certain layers.

**Pull request merged**